docs/standard/serialization/system-text-json-converters-how-to.md

```diff
@@ -0,0 +1,745 @@
+
+The following code shows a converter that converts `true` or `false` values in JSON to Boolean in an Object property, numbers to `long`, dates to `DateTime`, and strings to `string`. For other values, the code falls back to `JsonElement`.
+
+```csharp
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SystemTextJsonSamples
+{
+    public class ConverterInferredTypesToObject
+        : JsonConverter<object>
+    {
+        public override object Read(
+            ref Utf8JsonReader reader,
+            Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.True)
+            {
+                return true;
+            }
+
+            if (reader.TokenType == JsonTokenType.False)
+            {
+                return false;
+            }
+
+            if (reader.TokenType == JsonTokenType.Number)
+            {
+                if (reader.TryGetInt64(out long l))
+                {
+                    return l;
+                }
+
+                return reader.GetDouble();
+            }
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                if (reader.TryGetDateTime(out DateTime datetime))
+                {
+                    return datetime;
+                }
+
+                return reader.GetString();
+            }
+
+            // Use JsonElement as fallback.
+            // Newtonsoft uses JArray or JObject.
+            using (JsonDocument document = JsonDocument.ParseValue(ref reader))
+            {
+                return document.RootElement.Clone();
+            }
+        }
+
+        public override void Write(
+            Utf8JsonWriter writer,
+            object value,
+            JsonSerializerOptions options)
+        {
+            throw new InvalidOperationException("Should not get here.");
+        }
+    }
+}
+```
+
+The following code registers the converter:
+
+```csharp
+options = new JsonSerializerOptions();
+options.Converters.Add(new ConverterInferredTypesToObject());
+```
+
+Here's an example type with an Object property:
+
+```csharp
+public class WeatherForecastWithObjectProperties
+{
+    public object Date { get; set; }
+    public object TemperatureC { get; set; }
+    public object Summary { get; set; }
+}
+```
+
+The following example of JSON to deserialize contains values that will be deserialized as `DateTime`, `long`, and `string`:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot",
+}
+```
+
+Without the custom converter, deserialization puts a `JsonElement` in each property.
```

Comment:
However the TemperatureC property after deserialize will be a string (not the Temperature type).